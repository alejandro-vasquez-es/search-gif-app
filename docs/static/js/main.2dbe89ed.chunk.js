(this["webpackJsonpgift-app"]=this["webpackJsonpgift-app"]||[]).push([[0],{17:function(t,e,n){},18:function(t,e,n){"use strict";n.r(e);var a=n(1),c=n(7),r=n.n(c),i=n(2),s=n(9),o=n(0),u=function(t){var e=t.setCategories,n=Object(a.useState)(""),c=Object(i.a)(n,2),r=c[0],u=c[1];return Object(o.jsx)("form",{onSubmit:function(t){t.preventDefault(),r.trim().length>2&&(e((function(t){return[r].concat(Object(s.a)(t))})),u(""))},children:Object(o.jsx)("input",{type:"text",value:r,onChange:function(t){u(t.target.value)},placeholder:"Search your gif collection..."})})},j=n(10),d=function(t){t.id;var e=t.title,n=t.url;return Object(o.jsxs)("div",{className:"card  animate__animated animate__fadeIn",children:[Object(o.jsx)("img",{alt:e,src:n}),Object(o.jsx)("p",{children:e})]})},l=n(6),f=n.n(l),p=n(8),b=function(){var t=Object(p.a)(f.a.mark((function t(e){var n,a,c,r,i;return f.a.wrap((function(t){for(;;)switch(t.prev=t.next){case 0:return n="https://api.giphy.com/v1/gifs/search?q=".concat(encodeURI(e),"&limit=15&api_key=w8PQTRVGJpKfUGWhSd0Ev7Gj47jcVjCK"),t.next=3,fetch(n);case 3:return a=t.sent,t.next=6,a.json();case 6:return c=t.sent,r=c.data,i=r.map((function(t){var e;return{id:t.id,title:t.title,url:null===(e=t.images)||void 0===e?void 0:e.downsized_medium.url}})),t.abrupt("return",i);case 10:case"end":return t.stop()}}),t)})));return function(e){return t.apply(this,arguments)}}(),m=function(t){var e=t.category,n=function(t){var e=Object(a.useState)({data:[],loading:!0}),n=Object(i.a)(e,2),c=n[0],r=n[1];return Object(a.useEffect)((function(){b(t).then((function(t){r({data:t,loading:!1})}))}),[t]),setTimeout((function(){}),2e3),c}(e),c=n.data,r=n.loading;return Object(o.jsxs)(o.Fragment,{children:[Object(o.jsx)("h3",{children:e}),r&&Object(o.jsx)("p",{className:" animate__animated animate__flash",children:"Loading..."}),Object(o.jsx)("div",{className:"card-collection",children:c.map((function(t){return Object(o.jsx)(d,Object(j.a)({},t),t.id)}))})]})},h=function(){var t=Object(a.useState)(["Sailor Moon"]),e=Object(i.a)(t,2),n=e[0],c=e[1];return Object(o.jsxs)(o.Fragment,{children:[Object(o.jsx)("h2",{className:" animate__animated animate__fadeIn",children:"GifExpertApp"}),Object(o.jsx)(u,{setCategories:c}),Object(o.jsx)("hr",{}),Object(o.jsx)("ol",{children:n.map((function(t){return Object(o.jsx)(m,{category:t},t)}))})]})};n(17);r.a.render(Object(o.jsx)(h,{}),document.getElementById("root"))}},[[18,1,2]]]);
//# sourceMappingURL=main.2dbe89ed.chunk.js.map