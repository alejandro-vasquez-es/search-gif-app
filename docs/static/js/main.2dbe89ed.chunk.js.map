{"version":3,"sources":["components/AddCategory.js","components/GifCollectionItem.js","helpers/getGifts.js","components/GifCollection.js","hooks/useFetchGifs.js","GifExpertApp.js","index.js"],"names":["AddCategory","setCategories","useState","inputValue","setInputValue","onSubmit","e","preventDefault","trim","length","categs","type","value","onChange","target","placeholder","GifCollectionItem","id","title","url","className","alt","src","getGifs","category","a","encodeURI","fetch","response","json","data","gifs","map","gif","images","downsized_medium","GifCollection","loading","state","setState","useEffect","then","setTimeout","useFetchGifs","gift","GifExpertApp","categories","ReactDOM","render","document","getElementById"],"mappings":"uLAGaA,EAAc,SAAC,GAAqB,IAApBC,EAAmB,EAAnBA,cAEzB,EAAoCC,mBAAS,IAA7C,mBAAOC,EAAP,KAAmBC,EAAnB,KAcA,OACI,sBAAMC,SATW,SAACC,GAClBA,EAAEC,iBACCJ,EAAWK,OAAOC,OAAS,IAC1BR,GAAe,SAAAS,GAAM,OAAKP,GAAL,mBAAmBO,OACxCN,EAAc,MAKlB,SACI,uBACIO,KAAO,OACPC,MAAST,EACTU,SAjBc,SAACP,GACvBF,EAAcE,EAAEQ,OAAOF,QAiBfG,YAAc,qC,QCvBjBC,EAAoB,SAAC,GAAuB,EAArBC,GAAsB,IAAlBC,EAAiB,EAAjBA,MAAOC,EAAU,EAAVA,IAG3C,OACI,sBAAKC,UAAU,0CAAf,UACI,qBAAKC,IAAKH,EAAOI,IAAKH,IACtB,4BAAKD,Q,uBCRJK,EAAO,uCAAG,WAAOC,GAAP,uBAAAC,EAAA,6DACbN,EADa,iDACoCO,UAAWF,GAD/C,+DAEIG,MAAMR,GAFV,cAEbS,EAFa,gBAGEA,EAASC,OAHX,uBAGZC,EAHY,EAGZA,KAEDC,EAAOD,EAAKE,KAAI,SAAAC,GAAQ,IAAD,EACzB,MAAO,CACHhB,GAAIgB,EAAIhB,GACRC,MAAOe,EAAIf,MACXC,IAAG,UAAEc,EAAIC,cAAN,aAAE,EAAYC,iBAAiBhB,QATvB,kBAaZY,GAbY,4CAAH,sDCKPK,EAAgB,SAAC,GAAgB,IAAfZ,EAAc,EAAdA,SAE3B,ECJwB,SAACA,GAEzB,MAA0BtB,mBAAS,CAC/B4B,KAAM,GACNO,SAAS,IAFb,mBAAOC,EAAP,KAAcC,EAAd,KAwBA,OAnBAC,qBAAU,WAENjB,EAAQC,GAEHiB,MAAK,SAAAV,GAEFQ,EAAS,CACLT,KAAMC,EACNM,SAAS,SAKtB,CAACb,IAGJkB,YAAW,cACT,KAEKJ,EDtBwBK,CAAanB,GAAhCU,EAAZ,EAAOJ,KAAaO,EAApB,EAAoBA,QAEpB,OACI,qCACI,6BAAKb,IAEJa,GAAW,mBAAGjB,UAAY,oCAAf,wBAEZ,qBAAKA,UAAU,kBAAf,SAEQc,EAAOF,KAAI,SAACY,GAAD,OACP,cAAC,EAAD,eAEQA,GADIA,EAAK3B,aEf5B4B,EAAe,WAGxB,MAAoC3C,mBAAS,CAAC,gBAA9C,mBAAO4C,EAAP,KAAmB7C,EAAnB,KAEA,OACI,qCACI,oBAAImB,UAAY,qCAAhB,0BACA,cAAC,EAAD,CAAanB,cAAiBA,IAC9B,uBAEA,6BAEQ6C,EAAWd,KAAI,SAAAR,GAAQ,OACnB,cAAC,EAAD,CACIA,SAAYA,GACPA,Y,MCdjCuB,IAASC,OACL,cAAC,EAAD,IACAC,SAASC,eAAe,W","file":"static/js/main.2dbe89ed.chunk.js","sourcesContent":["import React, {useState} from 'react';\r\nimport PropTypes from 'prop-types';\r\n\r\nexport const AddCategory = ({setCategories}) => {\r\n\r\n    const [inputValue, setInputValue] = useState('');\r\n\r\n    const handleInputChange = (e) => {\r\n        setInputValue(e.target.value)\r\n    }\r\n\r\n    const handleSubmit = (e) =>{\r\n        e.preventDefault();\r\n        if(inputValue.trim().length > 2){\r\n            setCategories((categs => [inputValue,...categs]));\r\n            setInputValue('')\r\n        }\r\n    }   \r\n\r\n    return ( \r\n        <form onSubmit = {handleSubmit}>\r\n            <input \r\n                type = \"text\"\r\n                value = {inputValue}\r\n                onChange = {handleInputChange}\r\n                placeholder = 'Search your gif collection...'\r\n            />\r\n        </form>\r\n    )\r\n}\r\n\r\nAddCategory.propTypes= {\r\n    setCategories: PropTypes.func.isRequired\r\n}","import React from 'react'\r\n\r\nexport const GifCollectionItem = ( {id, title, url} ) => {\r\n\r\n\r\n    return (\r\n        <div className=\"card  animate__animated animate__fadeIn\">\r\n            <img alt={title} src={url} />\r\n            <p>{ title }</p>\r\n        </div>\r\n    )\r\n}\r\n","export const getGifs = async( category ) =>{\r\n    const url = `https://api.giphy.com/v1/gifs/search?q=${ encodeURI( category ) }&limit=15&api_key=w8PQTRVGJpKfUGWhSd0Ev7Gj47jcVjCK`;\r\n    const response = await fetch(url);\r\n    const {data} = await response.json();\r\n    \r\n    const gifs = data.map(gif => {\r\n        return {\r\n            id: gif.id,\r\n            title: gif.title,\r\n            url: gif.images?.downsized_medium.url\r\n        }\r\n    })\r\n    \r\n    return gifs\r\n}","import React from 'react'\r\nimport PropTypes from 'prop-types'\r\nimport { GifCollectionItem } from './GifCollectionItem.js';\r\nimport { useFetchGifs } from '../hooks/useFetchGifs.js';\r\n\r\nexport const GifCollection = ({category}) => {\r\n\r\n    const {data:images, loading} = useFetchGifs(category);\r\n    \r\n    return (\r\n        <>\r\n            <h3>{category}</h3>\r\n\r\n            {loading && <p className = ' animate__animated animate__flash'>Loading...</p>}\r\n\r\n            <div className='card-collection'>\r\n                {\r\n                    images.map((gift ) =>(\r\n                        <GifCollectionItem\r\n                            key = { gift.id }\r\n                            {...gift}\r\n                        />\r\n                ))}\r\n            </div>\r\n        </>\r\n    )\r\n}\r\n\r\nGifCollection.propTypes = {\r\n    category: PropTypes.string.isRequired\r\n}\r\n\r\n","import { useEffect, useState } from \"react\"\r\nimport { getGifs } from \"../helpers/getGifts.js\";\r\n\r\nexport const useFetchGifs = (category) =>{\r\n\r\n    const [state, setState] = useState({\r\n        data: [],\r\n        loading: true\r\n    });\r\n\r\n    useEffect(()=>{\r\n\r\n        getGifs(category)\r\n    \r\n            .then(gifs => {\r\n    \r\n                setState({\r\n                    data: gifs,\r\n                    loading: false\r\n\r\n                });    \r\n    \r\n        })  \r\n    }, [category])\r\n\r\n\r\n    setTimeout(()=>{\r\n    },2000)\r\n\r\n    return state\r\n}","import React, {useState} from 'react'\r\nimport { AddCategory } from './components/AddCategory.js';\r\nimport {GifCollection} from './components/GifCollection.js';\r\n\r\nexport const GifExpertApp = () => {\r\n\r\n    // const categories = ['One punch man', 'Tokyo ghoul', 'Dragon Ball']\r\n    const [categories, setCategories] = useState(['Sailor Moon']);\r\n\r\n    return (\r\n        <>\r\n            <h2 className = ' animate__animated animate__fadeIn'>GifExpertApp</h2>\r\n            <AddCategory setCategories = {setCategories}/>\r\n            <hr/>\r\n\r\n            <ol>\r\n                {\r\n                    categories.map(category => (\r\n                        <GifCollection\r\n                            category = {category}\r\n                            key={category}\r\n                        />\r\n                    ))\r\n                }\r\n            </ol>\r\n        </>\r\n    )\r\n}\r\n","import React from 'react';\r\nimport ReactDOM from 'react-dom';\r\nimport { GifExpertApp } from './GifExpertApp.js';\r\n\r\nimport './index.css';\r\n\r\nReactDOM.render(\r\n    <GifExpertApp / >,\r\n    document.getElementById('root')\r\n);"],"sourceRoot":""}