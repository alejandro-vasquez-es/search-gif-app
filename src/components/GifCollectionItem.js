import React from 'react'
import PropTypes from 'prop-types'


export const GifCollectionItem = ( {title, url} ) => {
    
    
    return (
        
        <div className="card animate__animated animate__fadeIn">
        <img alt={title} src={url} />
        <p>{ title }</p>
        </div>
        
        );
        
}
    
GifCollectionItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
    