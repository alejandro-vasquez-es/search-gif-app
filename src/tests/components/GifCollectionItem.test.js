import React from 'react';
import { shallow } from "enzyme";
import { GifCollectionItem } from "../../components/GifCollectionItem.js";

describe('<GifCollectionItem/> tests', () => {

    const url = 'https://localhost/algo.jpg';
    const title = 'regresa el tiempo y vuelve';
    const wrapper = shallow( < GifCollectionItem url = { url } title = { title } /> );

    test('should show correctly the component', () => {

        expect( wrapper ).toMatchSnapshot();

    });


    test('should have a paragraph with the title', () => {
        
        const p = wrapper.find('p')
        expect( p.text().trim() ).toBe( title )

    })

    test('should have the image with the url and alt attributes from the props', () => {

        const img = wrapper.find('img');
        // console.log( img.props() )
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    })
    
    test('should have animated fadeIn class', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className.includes( 'animate__fadeIn' ) ).toBe( true );

    })
    
    
    
});