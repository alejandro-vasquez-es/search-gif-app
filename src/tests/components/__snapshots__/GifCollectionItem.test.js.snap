// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifCollectionItem/> tests should show correctly the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="regresa el tiempo y vuelve"
    src="https://localhost/algo.jpg"
  />
  <p>
    regresa el tiempo y vuelve
  </p>
</div>
`;
